rules/framework.bzl

```diff
@@ -1191,6 +1191,7 @@ def _apple_framework_packaging_impl(ctx):
 
 apple_framework_packaging = rule(
     implementation = _apple_framework_packaging_impl,
+    toolchains = ["@bazel_tools//tools/cpp:toolchain_type"],
```

Comment:
Any downsides or fallout from adding this? I wondered if we need to add it to other rules; hmap, middlemen, etc

Reply:
I don't know of any. But I still haven't narrowed down the reasons for 👇 when using Bazel 7 with this change (like in https://github.com/bazel-ios/rules_ios/pull/797):
```
clang: error: using sysroot for 'AppleTVSimulator' but targeting 'MacOSX' [-Werror,-Wincompatible-sysroot]
```